sh2lib/sh2lib.c

```diff
@@ -246,10 +246,22 @@ int sh2lib_connect(struct sh2lib_config_t *cfg, struct sh2lib_handle *hd)
         .non_block = true,
         .timeout_ms = 10 * 1000,
     };
+
+#if ESP_IDF_VERSION_MAJOR > 4
+    hd->http2_tls = esp_tls_init();
+    if (!hd->http2_tls) {
+        ESP_LOGE(TAG, "Failed to allocate esp_tls handle!");
+        goto error;
+    }
+
+    if (esp_tls_conn_http_new_sync(cfg->uri, &tls_cfg, hd->http2_tls) != 1) {

```

Comment:
Please add comment that this is newly added API in place of previous `esp_tls_conn_http_new` in ESP-IDF v5.0